Implement AI commit message generation for Jetbrains

hassoncs opened the pull request:
This PR introduces a new Git commit message generation workflow for the JetBrains IDEs. It takes our existing logic and rearranges it so it's platform agnostic, so it can work for both VS Code and for JetBrains. 

It should be easiest to review commit by commit. The first commit rearranges the existing TypeScript code, and the second commit adds all of the Kotlin to support JetBrains specifically. 

Note: the Kotlin side is very heavily vibed since I'm not super familiar with Kotlin or those IDEs, but it seems reasonable to me! Any feedback is welcome

![2025-09-15 20 10 41](https://github.com/user-attachments/assets/8e143e51-75f4-43d2-b902-71b10be593e1)

SumJest commented:
Looking forward to this being part of the release! Any ETA for the merge?

hassoncs committed:
'chore: WIP'

hassoncs committed:
'chore: WIP'

cckenaaa commented:
I’m really excited about this feature and can’t wait to see it supported!

SumJest commented:
> I’m really excited about this feature and can’t wait to see it supported!

+1

hassoncs committed:
Merge remote-tracking branch 'origin/main' into jetbrains-git-commit-gen

hassoncs commented:
Merging this today! Just doing a final test pass now since it's been quite a while since this branch was created. Again, sorry for the delays!

hassoncs committed:
feat(jetbrains): add AI-powered commit message generation action

Add checkin handler factory and application service for commit message handling,
along with a new action in the VCS message group to generate AI-powered commit messages.
